crates/runtime/src/datafusion.rs

```diff
@@ -408,3 +415,28 @@ impl Default for DataFusion {
         Self::new()
     }
 }
+
+// TODO: This is a temporary solution since we cannot import secret and define this function in connection pool due to cyclical dependency
+// Ideally we want separate crate for secret and define this function in connection pool
+#[must_use]
+#[allow(clippy::implicit_hasher)]
+pub fn read_pg_config(
```

Comment:
This really doesn't belong here, we need to do this the right way.

I've done the secrets refactor, I'll leave the db_connection_pool crate to you.

https://github.com/spiceai/spiceai/pull/803